src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerExtensions.cs

```diff
@@ -95,7 +99,22 @@ public static IApplicationBuilder UseExceptionHandler(this IApplicationBuilder a
                 throw new ArgumentNullException(nameof(options));
             }
 
-            return app.UseMiddleware<ExceptionHandlerMiddleware>(Options.Create(options));
+            return app.Use(next =>
+            {
+                var loggerFactory = app.ApplicationServices.GetRequiredService<ILoggerFactory>();
+                var diagnosticListener = app.ApplicationServices.GetRequiredService<DiagnosticListener>();
+                var endpointDataSource = app.ApplicationServices.GetRequiredService<EndpointDataSource>();
+
+                if (!string.IsNullOrEmpty(options.ExceptionHandlingPath) && options.ExceptionHandler is null)
+                {
+                    var errorBuilder = app.New();
+                    errorBuilder.UseRouting(overrideEndpointRouteBuilder: false);
+                    errorBuilder.Run(next);
+                    options.ExceptionHandler = errorBuilder.Build();
+                }
+
+                return new ExceptionHandlerMiddleware(next, loggerFactory, Options.Create(options), diagnosticListener).Invoke;
```

Comment:
Why do you need to change this? Can't you use the normal `UseMiddleware<ExceptionHandlerMiddleware>`?